CONTRIBUTING.md

```diff
@@ -145,18 +148,112 @@ make check-pypi-packaging
 - Status checks SHOULD all be green.
   - Reviewer(s) have final word and HAVE TO run tests locally if they merge a PR with a red CI.
 
+#### Git workflow
+
+To keep our git history clean and easy to navigate we rebase our branches before
+merging as mentioned above. `pre-commit` checks if you need to rebase.
+
+For more info regarding rebasing and merging see:
+
+- [Atlassian tutorial for merging and rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing)
+- [Git - When to Merge vs. When to Rebase](https://derekgourlay.com/blog/git-when-to-merge-vs-when-to-rebase/)
+- [Git team workflows: merge or rebase?](https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase)
+
+##### Few tips regarding git workflow
+
+1. Write reasonable commit messages (See [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)).
+2. Try to think about the reviewer going through the pull request and commits.
+   For example don't make too many changes that aren't related, e.g. resolving
+   2 unrelated issues in one PR or having all changes in one commit.
+3. Rebasing lets you resolve any merge conflicts on your side. Also it allows CI
+   to check everything is passing, i.e. no mistakes were made by fixing conflicts.
+4. Since tests for `ogr` have very specific nature, it might be better to commit
+   response files in separate commit and then squash it before merging.
+5. Before merging you can polish commit messages if necessary or squash commits by rebasing.
+6. You can mention issues that will be closed in a PR description, e.g.:
+   `Closes #123`, `Fixes #123`; [more info](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords)
+
+##### Examples of git history
```

Comment:
That should not be needed thanks to the pre-commit check. But still a good example. Thanks